### DISCUSSION TITLE
Github Action gets killed on each invocation, would like to know why

### DISCUSSION THREAD

**Discussion by sanzgiri** [community/community#25402] [Actions]
<p>I have a github action in a public repo that runs some python code. Within this code, it downloads a dataset and starts training a simple linear regression model. The dataset is ~ 400 MB and the model is very basic. My guess is this is well within the disk usage and memory bounds.</p>
<p>However, the action gets killed on each invocation and all I see is the error code 137. I have logged in to the VM running the action using the tmate action. I see that the dataset is successfully downloaded and there is sufficient disk space.</p>
<p>How can I find out what is causing the action to get killed?</p>
<p>Here is a link to the latest failure of the action: <a href="https://github.com/sanzgiri/action-numerai/actions/runs/116260289" rel="nofollow noopener">https://github.com/sanzgiri/action-numerai/actions/runs/116260289</a></p>
<p>I would appreciate any help.</p>
<p>Thanks,<br>
Ashutosh</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/sanzgiri">@sanzgiri</a>,<br>
Based on my test, I found that the issue was caused by this line in <strong>train.py</strong>:</p>
<pre><code class="lang-auto">data = nx.download('numerai_dataset.zip')
</code></pre>
<p>It will first download the Numerai dataset and then load it. I think there is something wrong when load it.<br>
I would encourage you add load=False .</p>
<pre><code class="lang-auto">nx.download('numerai_dataset.zip', load=False)
</code></pre>

**Comment by sanzgiri** [comment]
<p>Thanks <a class="mention" href="/yanjingzhu">@yanjingzhu</a>. I was able to get the reason for the error after using a <code>ubuntu-16.04</code> runner</p>
<p><code>MemoryError: Unable to allocate 2.40 GiB for an array with shape (313, 2056947) and data type float32</code>.</p>
<p>Is there a way I can bump up the memory available to the runner? According to <a href="https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners" rel="nofollow noopener">https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners</a>, the VM has 7 GB of RAM. Is this accurate?</p>

**Comment by Yanjingzhu** [comment]
<p>Sorry to tell you that you could not increase the RAM memory of hosted runner. The workaround is to use a <a href="https://help.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners" rel="nofollow noopener">self-hosted runner</a>.<br>
But there are some security issue of using self-hosted runners with public repositories. Forks of your public repository can potentially run dangerous code on your self-hosted runner machine by creating a pull request that executes the code in a workflow.</p>
